### ISSUE TITLE
.POST json encoding

### ISSUE THREAD

**Issue by dreadjr** [MLSDev/TRON#14]
By default when i use the `.POST` method the `request.parameters` dict is encoded as url encoding. In the CRUD example shouldn't there be a `request.encoding = .JSON`, to force the body to be encoded as json. Or is there a better way of doing it?

**Comment by DenTelezhkin** [comment]
By default, Alamofire encodes parameters in URL for GET, HEAD and DELETE responses. For POST requests parameters are encoded in HTTP body. 

You are right, however, about JSON encoding, it is not forced, so if you don't specify .JSON encoding, HTTP body will have "application/x-www-form-urlencoded" Content-Type.

It is really interesting that this issue hasn't come up in our internal projects, seems like rails back-ends automatically consume both JSON and form-encoded HTTP bodies.

We will be looking into this issue for future releases, maybe we'll implement something like default encodings for HTTP method. 

Thanks for bringing this up!

**Comment by dreadjr** [comment]
Yeah I was wondering how you were using it internally. It's not a big deal
to add the encoding just had to dig in to see where the default was coming
from. The project is great, are you accepting pull requests if I come up
with anything?

**Comment by DenTelezhkin** [comment]
Basically, we never set encoding at all, it's always set to .URL =) Parameters then accept two kinds of parameters usage:

``` swift
parameters["foo"] = "bar"
parameters["dict[key]"] = "foo"
```

and 

``` swift
parameters["foo"] = "bar"
parameters["dict"] = ["key":"foo"]
```

I think first one will stop stop working once we use .JSON encoding. 

We welcome PR's, but it would be better if feature was discussed first on GitHub issue, so that you wouldn't be in situation where you spent some time, however feature was misaligned with project goals, and this time is basically wasted.

**Comment by DenTelezhkin** [comment]
I've addressed solution for this issue in 5fbb8a3f81496d4cabb8298c37d1d24e90e714e1.

This will be a part of 1.1.0 release that we'll be making soon. Again, thank you so much for bringing this up!

**Comment by dreadjr** [comment]
awesome, thanks.
